module/Photo/src/Photo/Service/Metadata.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Photo\Service;
+
+use Application\Service\AbstractService;
+use Zend\ServiceManager\ServiceManager;
+use Zend\ServiceManager\ServiceManagerAwareInterface;
+use Photo\Model\Album as AlbumModel;
+use Photo\Model\Photo as PhotoModel;
+
+/**
+ * Metadata service. This service implements all functionality related to 
+ * gathering metadata about photos.
+ */
+class Metadata extends AbstractService
+{
+
+    /**
+     * 
+     * @param \Photo\Model\Photo $photo the photo to add the metadata to
+     * @return \Photo\Model\Photo the photo with the added metadata
+     */
+    public function populateMetadata($photo, $path)
+    {
+        $exif = \read_exif_data($path, 'EXIF');
+        if (isset($exif['Artist'])) {
+            $photo->setArtist($exif['Artist']);
+        } else {
+            $photo->setArtist("Unknown"); //Needs to be t9n'd in the view
+        }
+        //I assume the exif data isn't deliberately stripped, so most values 
+        //are assumed to exist.
+        $photo->setCamera($exif['Model']);
+        $photo->setDateTime(\date_create($exif['DateTimeOriginal']));
+        $photo->setFlash($exif['Flash'] != 0);
+        $photo->setFocalLength($this->frac2dec($exif['FocalLength']));
+        $photo->setExposureTime($this->frac2dec($exif['ExposureTime']));
+        $photo->setShutterSpeed($this->exifGetShutter($exif));
+        $photo->setAperture($this->exifGetFstop($exif));
+        $photo->setIso($exif['ISOSpeedRatings']);
+        return $photo;
+    }
+
+    /*
+     * NOTE: Most code in the following part is copied from 
+     * the old site, mostly because I lack knowledge in photography.
+     */
+
+    /**
+     * Convert a string representing a rational number to a string representing 
+     * the corresponding decimal approximation. 
+     * @param string $str the rational number, represented as num+'/'+den
+     * @return float the decimal number, represented as float
+     */
+    private function frac2dec($str)
+    {
+        list($n, $d) = \explode('/', $str);
+        //Old site suppressed errors of previous line. No clue why.
```

Comment:
Dit gaat fout if de exposure time niet geset is. 

oplossing:
```
if (strpos($str, '/') === false) {
  return 0 // Or other value indicating it was not ok
}
```

daardoor is de if (!empty($d)) niet nodig. Eventueel kan er ook nog getest worden of $d gelijk is aan 0, en of $n en $d numeric zijn